pkg/executor/internal/mpp/local_mpp_coordinator.go

```diff
@@ -145,6 +145,9 @@ type localMppCoordinator struct {
 	enableCollectExecutionInfo bool
 	reportExecutionInfo        bool // if each mpp task needs to report execution info directly to coordinator through ReportMPPTaskStatus
 
+	// Record max task cnt among all the fragments.
+	// This is the number of nodes that involved in MPP computation.
+	maxTaskCnt int
```

Comment:
For sql with union, there may be `sender1(node1, node2) and sender2(node3, node4)`, so the node count will be task count of sender1 + task count of sender.

Reply:
from https://github.com/pingcap/tidb/pull/33084/files, the mpptask count must not be zero, seems `GetNodeCnt` can be removed.

Reply:
For union, `sender1(node1, node2) and sender2(node3, node4)`, because `sender1` and `sender2` will be in different fragment, so `coord.maxTaskCnt` is still `max(sender1_node_num, sender2_node_num)`, because we compute `c.maxTaskCnt` for each fragment.

Reply:
> This is the number of nodes that involved in MPP computation.
But `GetNodeCnt` means all nodes participating in mpp compute? So the node cnt here should be 4 instead of 2.

Reply:
It seems that `getNodeCnt` is only used to check whether `node_cnt` is equal to 0. However, an error has already been thrown for task_count being equal to 0 in https://github.com/pingcap/tidb/pull/33084/files.

Reply:
`GetNodeCnt()` is for update `MPPGather.nodeCnt`, so we can send http request with node info to AutoScaler.
And node cnt will be the hint of AutoScaler to scale out new CN

Reply:
> > This is the number of nodes that involved in MPP computation.
> 
> But `GetNodeCnt` means all nodes participating in mpp compute? So the node cnt here should be 4 instead of 2.

Yeah, you're right. nodeCnt should be 4

Reply:
ok, maybe the right `node_cnt = count(distinct node of scan executor)`

Reply:
I changed the implementation, using `map` to record all different nodes. ptal